src/com/facebook/buck/apple/AppleDescriptions.java

```diff
@@ -207,7 +211,18 @@ public static void populateCxxConstructorArg(
                     arg))
             .build();
 
-    output.srcs = arg.srcs;
+    if (arg.srcs.isPresent()) {
+      ImmutableSortedSet.Builder<SourceWithFlags> nonSwiftSrcs = ImmutableSortedSet.naturalOrder();
+      for (SourceWithFlags src: arg.srcs.get()) {
+        if (!MoreStrings.endsWithIgnoreCase(src.getSourcePath().toString(), SWIFT_EXTENSION)) {
```

Comment:
`MorePaths.getFileExtension(resolver.getAbsolutePath(src.getSourcePath()))` is safer